### PR TITLE
feat: Whitelist NFT collections & addresses

### PR DESCRIPTION
**Pull request by ikemHood**
Fixes: #86 

##Edited file `marketplace/marketplace.cairo`
- Modified `IMarketPlace` to include new traits.
- Added `whitelisted_collections` and `whitelisted_addresses` address to bool maps to `FlexDrop` contract storage
- Added `CollectionWhitelisted` and `AddressWhitelisted` events
- Added `is_collection_whitelisted` and `is_address_whitelisted` getters
- Added `whitelist_collection` and `whitelist_address` setters
- Modified `calculate_protocol_fee` and `transfer_fees_and_funds` to accomodate update.

### CODE REVIEW

**Comment by ikemHood** [comment]
Hey @0xandee , Please review.

is it neccesary to add a `exempt_whitelist_from_fee` state to allow enabling and disabling whitelist exemption from fees?

**Comment by 0xandee** [comment]
Hey @ikemHood , #86 supposed to developed on `marketplace` contracts, not `openedition` 🙏 Would you mind work in that contracts?

**Comment by ikemHood** [comment]
Alright.

**Comment by 0xandee** [comment]
Yes I believe so!

**Comment by ikemHood** [comment]
@0xandee  please review now...

File: marketplace/src/marketplace.cairo [hunk 1]
```diff
@@ -46,6 +46,8 @@ trait IMarketPlace<TState> {
     fn update_royalty_fee_manager(ref self: TState, manager: ContractAddress);
     fn update_transfer_selector_NFT(ref self: TState, selector: ContractAddress);
     fn update_signature_checker(ref self: TState, checker: ContractAddress);
+    fn whitelist_collection(ref self: TState, collection: ContractAddress, is_whitelisted: bool);
+    fn whitelist_address(ref self: TState, address: ContractAddress, is_whitelisted: bool);
```
[the remark this answers is not in the document]
Comment (ikemHood) [reply marketplace/src/marketplace.cairo]: changes in openedition files now are formating from running scarb fmt after removing the previos changes

[the remark this answers is not in the document]
Comment (0xandee) [reply marketplace/src/marketplace.cairo]: Cool, have you deploy and test if it works as planned?

[the remark this answers is not in the document]
Comment (0xandee) [reply marketplace/src/marketplace.cairo]: Hey do you have any estimation on this issue @ikemHood

[the remark this answers is not in the document]
Comment (ikemHood) [reply marketplace/src/marketplace.cairo]: the implememntation is complete,

I would be performing test on it. should do that before EOD

[the remark this answers is not in the document]
Comment (0xandee) [reply marketplace/src/marketplace.cairo]: Got ya!

**Pull request merged**